src/scripts/data_processing/generate_data/generate.py

```diff
@@ -266,10 +254,10 @@ def run(self, args, logger, metrics_logger, unknown_args):
 
         # record a metric
         logger.info(f"Generating data.")
-        if args.type == "classification":
-            self.generate_classification(args)
-        elif args.type == "regression":
-            self.generate_regression_tasks(args)
+        if args.type in ["regression", "classification", "lambdarank"]:
```

Comment:
now that all 3 types are consistent, you can even skip this test and leave it to generate_task() to figure it out, we can even remove NotImplementedError since argparse will filter out the wrong values for us (I forgot that in my initial code).

Reply:
done